### DISCUSSION TITLE
webRTC connection

### DISCUSSION THREAD

**Discussion by ANNAMALAI357** [Ylianst/MeshCentral#5552] [Q&A]
Hi, I am running my MC server in a Windows VM. 
I have enabled webRTC by :
```
"WebRTC": true
"WebRtConfig": {
  "iceServers": [
    {
      "urls": "stun:stun.l.google.com:19302"
    },
    {
      "urls": "stun:stun.services.mozilla.com"
    }
  ]
}
```
And I could see webRTC connected text once I click connect to any of my node.
But still it is not connecting through webRTC. So maybe I might be missing some detail here.
In the discussion, it was told webRTC uses UDP to communicate. so Do I have to enable UDP communication in any of the ports in firewall in my VM ? 
can someone help me out !!!

**Comment by NiceGuyIT** [comment]
While I'm not familiar with MeshCentral's WebRTC implementation, most implementations use hole punching and thus do not require a dedicated port and corresponding firewall rule.

Have you turned on debug logging on the client and server to see if there are any messages about not being able to establish a WebRTC connection?

Is the agent, server and your browser on the same LAN? Are you going over the WAN? If WAN is used, does your ISP use CGNAT? CGNAT can cause problems with hole punching.

> But still it is not connecting through webRTC.

How have you determined this? Did you analyze a TCP dump showing all traffic is routing through the server?

What version is the server? What version is the agent?

Have you tried an incognito/private window to make sure any browser plugins/configuration aren't playing a factor?
